### PR TITLE
Implement external strings.

### PR DESCRIPTION
**Pull request by zherczeg**

### CODE REVIEW

**Comment by ossy-szeged** [comment]
It introduces a new public API, please add documentation for it, maybe an example would be great too.

File: docs/02.API-REFERENCE.md [hunk 1]
```diff
@@ -4564,6 +4564,96 @@ jerry_create_string_sz_from_utf8 (const jerry_char_t *str_p,
...
+  jerry_value_t string_value  = jerry_create_external_string ((const jerry_char_t *) char_array,
+                                                              NULL);
+
+  ... // usage of string_value
+
+  jerry_release_value (string_value);
+}
+```
+
+**See also**
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md:4604]: Please add this note below: `*New in version [[NEXT_RELEASE]]*.`

File: docs/02.API-REFERENCE.md [hunk 2]
```diff
@@ -4564,6 +4564,96 @@ jerry_create_string_sz_from_utf8 (const jerry_char_t *str_p,
...
+                                                                 NULL);
+
+  ... // usage of string_value
+
+  jerry_release_value (string_value);
+}
+
+```
+
+**See also**
```
Comment (rerobika) [inline docs/02.API-REFERENCE.md:4651]: Ditto.

File: docs/02.API-REFERENCE.md [hunk 3]
```diff
@@ -4564,6 +4564,100 @@ jerry_create_string_sz_from_utf8 (const jerry_char_t *str_p,
...
+- return value - value of the created string
+
+*New in version [[NEXT_RELEASE]]*
+
+**Example**
+
+```c
+{
+  const char* string_p = "a large and immutable string: this is a story about ....";
+  jerry_value_t string_value  = jerry_create_external_string ((const jerry_char_t *) char_array,
```
Comment (dbatyai) [inline docs/02.API-REFERENCE.md:4597]: char_array -> string_p

File: docs/02.API-REFERENCE.md [hunk 4]
```diff
@@ -4564,6 +4564,100 @@ jerry_create_string_sz_from_utf8 (const jerry_char_t *str_p,
...
+- return value - value of the created string
+
+*New in version [[NEXT_RELEASE]]*
+
+**Example**
+
+```c
+{
+  const char* string_p = "a large and immutable string: this is a story about ....";
+  jerry_value_t string_value  = jerry_create_external_string_sz ((const jerry_char_t *) char_array,
```
Comment (dbatyai) [inline docs/02.API-REFERENCE.md:4644]: char_array -> string_p

**Comment by zherczeg** [comment]
Thank you for the review, patch is updated.

**Pull request merged**